cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/FieldLineageRelations.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.metadata;
+
+import io.cdap.cdap.proto.id.DatasetId;
+
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * This class represents an aggregation of the field level lineage about an incoming/outgoing dataset. The entityId
+ * represents the dataset. And the relations contains all the field level lineage .
+ */
+public class FieldLineageRelations {
+  private final DatasetId entityId;
+  private final Set<FieldRelation> relations;
```

Comment:
So each field lineage relationship is about two fields and two dataset, right? Why we only have one datasetId stored at the relations level, and the `FieldRelation` doesn't contain any information about the other dataset?

Reply:
This class should be used with the `DatasetFieldLineageSummary` as the summary will contain the information about the other dataset. I have moved this class as an inner class